### PR TITLE
bug/5291-error-when-handling-empty-ottrk-files 

### PR DESCRIPTION
**Pull request by randy-seng**
This pull request addresses handling empty data scenarios in `track_store.py` and `pandas_parser.py` to ensure proper initialization of `PandasTrackDataset`. It includes logic enhancements and new tests to validate correct column and index setup, preventing potential issues in edge cases.

OP#5291

### CODE REVIEW

File: tests/OTAnalytics/plugin_datastore/test_track_store.py [hunk 1]
```diff
@@ -637,3 +639,15 @@ def test_create_test_flyweight_with_single_detection(
...
+        """
+        #Bugfix https://openproject.platomo.de/projects/001-opentrafficcam-live/work_packages/5291
+
+        @bug by randy-seng
+        """  # noqa
+        target = PandasTrackDataset(track_geometry_factory=Mock())
+        actual = target.get_data()
+        assert actual.index.names == INDEX_NAMES
+        assert INDEX_NAMES not in actual.columns.to_list()
+        assert set(COLUMNS) - set(INDEX_NAMES) - set(actual.columns.to_list()) == set()
```
Comment (schubsen) [inline tests/OTAnalytics/plugin_datastore/test_track_store.py:653]: assert set(COLUMNS) - set(INDEX_NAMES) == set(actual.columns.to_list())
would be better, since it also checks whethere there are columns in the actual that don't appear in COLUMNS and INDEX

**Review by schubsen — CHANGES_REQUESTED** [review]

**Review by schubsen — APPROVED** [review]

**Review by schubsen — APPROVED** [review]
